### PR TITLE
Refactor pool transactions

### LINKED ISSUE #671: Fetch pool transactions from API instead of local storage
Before, there was no API to get the list of all transactions in the pool, so we stored them in storage and fetched them 1 by 1. Now we do have an api for this and we should use it.

### PR DESCRIPTION
**Pull request by elias-garcia**
Closes #671.

### What does this PR does?

This PR uses the new `getPoolTransactions` from the `CoordinatorAPI` instead of the `TxPool` which was using the `localStorage`. It also moves the `poolTransactionsTask` to the `global` state instead of having it duplicated in almost all of the modules of the application.

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Respect code style and lint
- [x] Update documentation (if needed)

### CODE REVIEW

**Comment by amonsosanz** [comment]
I'm still getting a compilation error after doing `rm -rf node_modules/ build/`:

```
Failed to compile.

../hermezjs/src/signers/ledger-signer.js
Module not found: Can't resolve '@ledgerhq/hw-app-eth' in '/home/alex/dev/hermez/hermezjs/src/signers'
```

File: src/types/hermezjs.d.ts [hunk 1]
```diff
@@ -290,7 +295,6 @@ declare module "@hermeznetwork/*" {
   export { default as Tx } from "@hermeznetwork/hermezjs/src/tx";
   export { default as TxUtils } from "@hermeznetwork/hermezjs/src/tx-utils";
   export { default as TxFees } from "@hermeznetwork/hermezjs/src/tx-fees";
-  export { default as TxPool } from "@hermeznetwork/hermezjs/src/tx-pool";
```
[the remark this answers is not in the document]
Comment (elias-garcia) [reply src/types/hermezjs.d.ts:293]: This module does not exist anymore if this PR https://github.com/hermeznetwork/hermezjs/pull/227 is approved in `hermezjs`.

File: src/types/hermezjs.d.ts [hunk 2]
```diff
@@ -490,21 +493,6 @@ declare module "@hermeznetwork/hermezjs/src/tx-fees" {
   ): Promise<number>;
 }
 
-// TxPool
-declare module "@hermeznetwork/hermezjs/src/tx-pool" {
```
[the remark this answers is not in the document]
Comment (elias-garcia) [reply src/types/hermezjs.d.ts:494]: Same as before, it's not going to exist anymore!

File: src/adapters/hermez-api.ts [hunk 3]
```diff
@@ -318,15 +317,23 @@ export function fetchHermezAccount(
...
+  return CoordinatorAPI.getPoolTransactions(
+    hermezEthereumAddress,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    limit
+  ).then((poolTxs) => poolTxs.transactions);
```
[the remark this answers is not in the document]
Comment (elias-garcia) [reply src/adapters/hermez-api.ts]: Agree, fixed!

File: src/types/hermezjs.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/types/hermezjs.d.ts:293]: Aha! Cool then, I still have to review that PR.

File: src/views/account-details/account-details.view.tsx [hunk 4]
```diff
@@ -338,11 +338,14 @@ function AccountDetails({
               const tokenTimerWithdraws = accountTimerWithdraws.filter(
                 (withdraw) => withdraw.token.id === accountTask.data.token.id
               );
+              const accountPoolTransactions = poolTransactionsTask.data.filter(
+                (tx) => tx.fromAccountIndex === accountIndex
+              );
```
[the remark this answers is not in the document]
Comment (elias-garcia) [reply src/views/account-details/account-details.view.tsx:343]: Right, good catch! I've fixed it. Now we can get them in order from the API.

File: src/store/global/global.reducer.ts [hunk 5]
```diff
@@ -162,6 +167,33 @@ function globalReducer(
...
+          data: action.transactions,
+        },
+      };
+    }
+    case GlobalActionTypes.LOAD_POOL_TRANSACTIONS_FAILURE: {
+      return {
+        ...state,
+        poolTransactionsTask: {
+          status: "failed",
+          error: "An error ocurred loading the transactions from the pool",
```
[the remark this answers is not in the document]
Comment (elias-garcia) [reply src/store/global/global.reducer.ts:193]: In this PR we don't have decoding so we don't know why the function call has failed. I'll do it after merging it with the other branch.

[the remark this answers is not in the document]
Comment (amonsosanz) [reply src/store/global/global.reducer.ts:193]: As agreed, we better merge this branch first and then I'll do it in my branch for this and other manual errors like this one.

**Review by amonsosanz — APPROVED** [review]

**Review by AlbertoElias — APPROVED** [review]

**Pull request merged**